src-electron/upgrade/upgrade.js

```diff
@@ -22,6 +22,13 @@ const dbEnum = require('../../src-shared/db-enum.js')
 const fs = require('fs')
 const fsp = fs.promises
 
+async function parseJson(json) {
+  try {
+    return JSON.parse(json)
+  } catch (err) {
+    return undefined
```

Comment:
? You are hiding exception here. Why do we need this? This feels wrong. I don't want for people to blindly start using this function, and then be confused why they get "error: undefined", instead of "error: JSON parsing error".

So this function makes no sense to me.